gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java

```diff
@@ -306,10 +305,10 @@ public Single<Certificate> update(String domain, String id, UpdateCertificate up
                 })
                 .map(this::getValid)
                 .flatMap(certificate -> certificateRepository.update(certificate))
-                // create event for sync process
-                .flatMap(certificate1 -> {
-                    Event event = new Event(Type.CERTIFICATE, new Payload(certificate1.getId(), ReferenceType.DOMAIN, certificate1.getDomain(), Action.UPDATE));
-                    return eventService.create(event).flatMap(__ -> Single.just(certificate1));
+                .doOnSuccess(certificate -> {
+                    auditService.report(AuditBuilder.builder(CertificateAuditBuilder.class).principal(principal).referenceId(domain).referenceType(ReferenceType.DOMAIN).type(EventType.CERTIFICATE_UPDATED).certificate(certificate));
+                    Event event = new Event(Type.CERTIFICATE, new Payload(certificate.getId(), ReferenceType.DOMAIN, certificate.getDomain(), Action.UPDATE));
+                    eventService.create(event).subscribe();
```

Comment:
**question**: do we really want to fire&forget the event creation ? :thinking: If an error occures the event may not be persist in DB and so the GW will never see the change. Before this update, an error will be display on the UI, after this change, the UI will display a successful response but the GWs will not see the updates...